File: app/src/main/java/android/kotlin/elements/_1_simple_java_recycler_view/SimpleAdapter.java
package android.kotlin.elements._1_simple_java_recycler_view;

import android.content.Context;
import android.kotlin.elements.R;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class SimpleAdapter extends RecyclerView.Adapter<SimpleViewHolder> {
    //Список элементов
    List<String> strings;

    //количество ViewVholders, которых будут переиспользоваться
    private static int viewHolderCount;
    private Context parent;

    public SimpleAdapter(List<String> strings, Context parent) {
        this.strings = strings;
        this.parent = parent;
        viewHolderCount = 0;
    }

    //Данынй метод будет вызываться меньше, чем число элементов.
    //Каждый раз при вызове данного метода - будет создаваться новый ViewHolder.
    //Затем они будут переиспользоваться.

    @NonNull
    @Override
    public SimpleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        Context context = parent.getContext();

        //LayoutInflater - это класс который позволяет из xml создавать новые представления
        LayoutInflater inflater = LayoutInflater.from(context);

        //parent - родительский элемент. false - необяхательный параметр. Говорит нужно ли созданный элемент помещать внутрь родителя. У нас parent
        //это RecyclerView - он и сам это сделает
        View view = inflater.inflate(R.layout.simple_java_recycler_view_item, parent, false);

        SimpleViewHolder simpleViewHolder = new SimpleViewHolder(view);

        simpleViewHolder.indexTV.setText("ViewHolder indexTV: " + viewHolderCount);

        viewHolderCount++;

        return simpleViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull SimpleViewHolder holder, int position) {
        holder.bind(strings.get(position));
    }

    @Override
    public int getItemCount() {
        return strings.size();
    }
}

File: app/src/main/java/android/kotlin/elements/simple_kotlin_recycler_view/SimpleAdapter.kt
package android.kotlin.elements.simple_kotlin_recycler_view

import android.content.Context
import android.kotlin.elements.R
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class SimpleAdapter(val strings:List<String>, val parent:Context): RecyclerView.Adapter<SimpleViewHolder>() {

    companion object{
        var viewHolderCount:Int = 0
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SimpleViewHolder {
        val context:Context = parent.context
        val inflater:LayoutInflater = LayoutInflater.from(context)

        var view: View = inflater.inflate(R.layout.simple_java_recycler_view_item,parent,false)

        var simpleViewHolder:SimpleViewHolder = SimpleViewHolder(view)

        simpleViewHolder.indexTV.setText("ViewHolder indexTV: " + viewHolderCount)

        viewHolderCount++

        return simpleViewHolder

    }

    override fun onBindViewHolder(holder: SimpleViewHolder, position: Int) {
        holder.bind(strings.get(position))
    }

    override fun getItemCount(): Int {
        return strings.size
    }
}